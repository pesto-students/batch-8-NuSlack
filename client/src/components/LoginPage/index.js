import React, { useState } from 'react';
import styled from 'styled-components';
import { Input, Button } from 'antd';
import { Redirect } from 'react-router-dom';
import axios from 'axios';
import { useHomeContext } from '../../context/HomeContext';

const LoginContainer = styled.div`

display: flex;
justify-content: center;
align-items: center;
height: 100vh;
flex-direction: column;
input{
    width: 20em;
}
button{
    margin-top: 1em;
}
`;
const LoginPage = () => {
  const [input, setInput] = useState('');
  const { user, setUser } = useHomeContext();
  const [loggedIn, setLoggedIn] = useState(false);
  const onInputChange = (e) => {
    setInput(e.target.value);
  };
  const updateStateAndRedirect = async () => {
    setUser({ username: input });
    const userDataResponse = await axios.get('http://localhost:8080/users', {
      params: {
        username: input,
      },
    });

    if (userDataResponse.data[0]) {
      setUser({ ...userDataResponse.data[0] });
      setLoggedIn(true);
    }
  };

  if (loggedIn || (user && user._id)) {
    return <Redirect to="/home" />;
  }
  return (
    <LoginContainer>
      <h1>Login</h1>
      <Input value={input} onChange={onInputChange} onPressEnter={updateStateAndRedirect} />
      <Button onClick={updateStateAndRedirect}>Login</Button>
    </LoginContainer>
  );
};

export default LoginPage;
