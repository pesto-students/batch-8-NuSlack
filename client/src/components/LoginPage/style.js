import styled from 'styled-components';

const LoginContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  flex-direction: column;
  .signup-logo {
    @media only screen and (max-width: 600px) {
      height: 175px;
    }
  }
`;

const anotherStyle = {};

export { LoginContainer, anotherStyle };
