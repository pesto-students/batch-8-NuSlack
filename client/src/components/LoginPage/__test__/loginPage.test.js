import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import LoginPage from '../index';

describe('<LoginPage />', () => {
  it('should match the snapshot', () => {
    const tree = shallow(<LoginPage />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
