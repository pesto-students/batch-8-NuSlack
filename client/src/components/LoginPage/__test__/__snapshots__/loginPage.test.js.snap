// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> should match the snapshot 1`] = `
<styled.div>
  <img
    alt="logo image"
    src="/images/logo.png"
  />
  <h1>
    Login to NuSlack
  </h1>
  <Form(LoginForm)
    handleFormSubmit={[Function]}
  />
  <div>
    OR
  </div>
  <e
    accessType="online"
    buttonText="Login with Google"
    clientId="478468868101-i9uk561c3coju59vdtjir6hipo17v6m8.apps.googleusercontent.com"
    cookiePolicy="single_host_origin"
    disabledStyle={
      Object {
        "opacity": 0.6,
      }
    }
    fetchBasicProfile={true}
    icon={true}
    isSignedIn={false}
    jsSrc="https://apis.google.com/js/api.js"
    onFailure={[Function]}
    onRequest={[Function]}
    onSuccess={[Function]}
    prompt=""
    scope="profile email"
    tag="button"
    theme="light"
    type="button"
    uxMode="popup"
  />
</styled.div>
`;
